source/client/HandManagement.py

```diff
@@ -0,0 +1,114 @@
+import pygame
+import client.Button as Btn
+import client.UIConstants as UIC
+from client.UICardWrapper import UICardWrapper
+
+"""This file contains methods used in displaying, selecting and sorting the hand"""
+
+
+def wrapHand(self, updated_hand, wrapped_hand):
+    """Associate each card in updated_hand with a UICardWrapper
+
+    if change is that new cards were added, then want to preserve location and status for cards that were
+    already in hand.
+    If cards were played, then the wrapped cards that are removed should to those cards with status = 2
+    (prepared cards).
+    Note that discards are removed immediately after controller confirms discard legal.
+    """
+    card_xy = (10, UIC.Table_Hand_Border + 40)
+    # sort cards so that if prepared cards were played, those are the instances of the cards that are removed.
+    old_wrapped_hand = sorted(wrapped_hand, key=lambda x: x.status)
+    updated_wrapped_hand = []
+    if not updated_hand == []:
+        for card in updated_hand:
+            newcard = True
+            for already_wrapped in old_wrapped_hand:
+                if newcard and card == already_wrapped.card:
+                    card_wrapped = already_wrapped
+                    card_xy = (max(card_xy[0], card_wrapped.img_clickable.x), card_xy[1])
+                    old_wrapped_hand.remove(already_wrapped)
+                    newcard = False
+            if newcard:
+                card_xy = (card_xy[0] + self.hand_scaling[1], card_xy[1])
+                card_wrapped = UICardWrapper(card, card_xy, self.hand_scaling[0])
+            updated_wrapped_hand.append(card_wrapped)
+        maxX = UIC.Disp_Width - (self.hand_scaling[1] / 2)
```

Comment:
A line gap and a comment about moving on to scaling/rendering should probably go here.

Reply:
Put in some comments.